src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -75,7 +75,9 @@ public ReadOnlyTaskList getTaskList() {
 
     @Override
     public ObservableList<Task> getFilteredTaskList() {
-        return model.getFilteredTaskList();
+        ObservableList<Task> tasklist = model.getFilteredTaskList();
+        tasklist.forEach((task) -> task.triggerRecurringIfPresent(model));
```

Comment:
should use storage.saveTaskList here instead of inside Task.java's timerTask

Reply:
Also maybe this isn't the best place to triggerRecurring, it should be done when you initalize the tasklist and also when you add a new task/ edit task meaning whenever a new task is created